### DISCUSSION TITLE
Welcome to django-pint Discussions!

### DISCUSSION THREAD

**Discussion by CarliJoy** [CarliJoy/django-pint#26] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by CarliJoy** [comment]
@cornelv finally "here" gitlab opened a place to chat.
Could you find some time to do something? 
Don't worry if not :-)
But tell a bit in advance if you plan to code something, I might refractor some stuff...

**Reply by cornelv** [reply]
@CarliJoy yes, this section is very helpful :)  
unfortunately I didn't find the time to work on the tasks yet, I was planing to do over the upcoming weekend. If you want to change the task list is fine by me.

**Reply by cornelv** [reply]
also, by any change do you have a local environment setup guide or something similar ? it would help me get started faster. thank you

**Reply by CarliJoy** [reply]
Dear cornelv,

currently I have a bit of a headache therefore just a little list of commands with no guarantee for correctness.
BTW: I assume you develop under Linux, if you have windows you have to research stuff yourself

The instructions assume that your python is python3

1. Install postgres on your computer (i.e. `apt-get install postgresql`)
1.  Delete and recreate your forked repro
1.  Clone your forked repro locally
1.  Checkout the current improve_testing branch and setup postgres as described at the end of the readme
1.  Install pre-commit i.e. with `pip install pipx && pipx install pre-commit` (Works also with pip but pipx creates virtual environments directly)
1.  activate pre-commit so you commit only well formatted thing `pre-commit install`
1.  python -m .venv && source .venv/bin/activate.sh
1.  Prepare build environment `pip install --upgrade pip wheel setuptools`
1.  Locally install the package with all testing requirements `pip install -e '.[testing]'`
1.  Develop :-)
1.  Optional activate PGP signing for every commit: `git config commit.gpgsign true `
1.  before pushing run `pytest` locally to see if the issues are solved.


Once you set up your dev environment you could take the listed step and add them to the documentation ;-)

**Reply by cornelv** [reply]
Hi CarliJoy, thank you for the detailed information, I unfortunately I wasn't able to install the local environment on my local machine, I had some issues with python version on osx.
I created a dockerfile and a docker-compose file that creates the local environment and I added instructions for it in the readme file. I am not sure if you want to go this route so I created a PR to my fork only so you can check it first: https://github.com/cornelv/django-pint/pull/1/files

let me know what do you think.

thank you
